src/test/java/org/broadinstitute/hellbender/utils/spark/SparkUtilsUnitTest.java

```diff
@@ -69,6 +84,157 @@ public void testPathExists() throws Exception {
+        JavaRDD<GATKRead> pairedReads = SparkUtils.putReadsWithTheSameNameInTheSamePartition(header, reads, ctx);
+        List<List<GATKRead>> partitions = pairedReads.mapPartitions((FlatMapFunction<Iterator<GATKRead>, List<GATKRead>>) it ->
+                Iterators.singletonIterator(Lists.newArrayList(it))).collect();
+        assertEquals(partitions.size(), numPartitions);
+        for (int i = 0; i < numPartitions; i++) {
+            assertEquals(partitions.get(i).size(), expectedReadsPerPartition[i]);
+        }
+        assertEquals(Arrays.stream(expectedReadsPerPartition).sum(), numPairs * numReadsInPair);
+    }
+
+    private static List<GATKRead> createPairedReads(SAMFileHeader header, int numPairs, int numReadsInPair) {
+        final int readSize = 151;
+        final int fragmentLen = 400;
+        final String templateName = "readpair";
+        int leftStart = 10000;
+        List<GATKRead> reads = new ArrayList<>();
+        for (int i = 0; i < numPairs;i++) {
+            leftStart += readSize * 2;
+            int rightStart = leftStart + fragmentLen - readSize;
+            reads.addAll(ArtificialReadUtils.createPair(header, templateName + i, readSize, leftStart, rightStart, true, false));
+            // Copying a secondary alignment for the second read to fill out the read group
+            GATKRead readToCopy = reads.get(reads.size()-1).copy();
+            readToCopy.setIsSecondaryAlignment(true);
+            for (int j = 2; j < numReadsInPair; j++) {
+                reads.add(readToCopy.copy());
+            }
+        }
+        return reads;
+    }
+
+    @Test(expectedExceptions = GATKException.class)
+    public void testReadsPairsSpanningMultiplePartitionsCrash() {
+        JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader();
+        header.setSortOrder(SAMFileHeader.SortOrder.queryname);
+        List<GATKRead> reads = createPairedReads(header, 40, 2);
+        // Creating one group in the middle that should cause problems
+        reads.addAll(40, createPairedReads(header, 1, 30));
+
+        JavaRDD<GATKRead> problemReads = ctx.parallelize(reads,5 );
+        SparkUtils.putReadsWithTheSameNameInTheSamePartition(header, problemReads, ctx);
+    }
+
+    @Test
+    public void testReadsMustBeQueryGroupedToFixPartitions(){
+        JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader();
+        header.setSortOrder(SAMFileHeader.SortOrder.coordinate);
+        List<GATKRead> reads = createPairedReads(header, 40, 2);
+        final JavaRDD<GATKRead> readsRDD = ctx.parallelize(reads, 5);
+        Assert.assertThrows(IllegalArgumentException.class, () -> SparkUtils.putReadsWithTheSameNameInTheSamePartition(header, readsRDD, ctx));
+    }
+
+    @Test
+    public void testSortCoordinateSort() {
```

Comment:
rename to reflect that it is asserting equivalence to htsjdk coordinate sort (or add comment)

Reply:
done for both